test/runnable/test20714.d

```diff
@@ -0,0 +1,22 @@
+// https://issues.dlang.org/show_bug.cgi?id=20714
+
+struct Blitter
+{
+    int payload;
+    this(this){}
+}
+
+struct Adder
+{
+    Blitter blitter;
+    this(int payload) {this.blitter.payload = payload;}
+    this(ref Adder rhs) {this.blitter.payload = rhs.blitter.payload + 1;}
+}
```

Comment:
One way to  make this more user friendly would be to print a deprecation message asking to explicitly disable the postblot in the parent struct.

Reply:
I thought about that but making your struct non-copyable just because you want to use an instance of it as a field in a struct that has a copy constructor is not tractable (think of library code). Also there are cases like:

```d
struct Blitter
{
    this(this) {}
}

struct Copy
{
    Blitter b;
    this(ref Copy c)
    {
         this.b = c.b;          // -> rely on postblit call
     }
}
```

Reply:
One warning would work though: "Careful, you have fields with postblits! There will be no automatic generation of postblit for struct Copy and copy constructor will take precedence"

Reply:
What I meant was this:
```
struct Blitter
{
    this(this) {}
}

struct Copy
{
    Blitter b;
    @disable this(this); // If not present, deprecation message
    this(ref Copy c)
    {
         this.b = c.b;          // -> rely on postblit call
     }
}
```

This would work for library code, and avoid the silent change in behavior.

Reply:
From this comment [1]: "I don't want to further infest the codebase with postblits"

[1] https://issues.dlang.org/show_bug.cgi?id=20965#c3

Reply:
Well their code-base already uses postblit, event if that postblit comes from library code.
This changes breaks library code that relies on postblit, e.g.:
```
struct Foo
{
    int a;

    this(this)
    {
     	this.a++;
    }
}

struct Bar
{
    Foo f;
    int b;
    this(ref typeof(this) c)
    {
     	this.b++;
    }
}

void main ()
{
    Bar b;
    assert(b.f.a == 0);
    Bar c = b;
    assert(c.f.a == 1);
}
```